### PR TITLE
Fix org and space role association

### PR DESCRIPTION
**Pull request by jcscottiii**
### First commit: Fix functional test server to check for correct roles
This modification makes sure that all requests to do
association/dissociation of roles use the right api key for the role.

This will break tests because currently the code doesn't use the right
api key for role modification but will be fixed in following commits.

### Other commits: fixing the api calls and tests

### CODE REVIEW

File: static_src/stores/user_store.js [hunk 1]
```diff
@@ -164,6 +150,7 @@ export class UserStore extends BaseStore {
         const orgPermissionsReq = cfApi.deleteOrgUserPermissions(
           action.userGuid,
           action.orgGuid,
+          'users',
```
Comment (rememberlenny) [inline static_src/stores/user_store.js]: @jcscottiii This might be a duplicated line

Comment (rememberlenny) [inline static_src/stores/user_store.js]: https://github.com/18F/cg-dashboard/blob/0ed527b4fe52d824abe738d9b885a0ef0256c6d7/static_src/stores/user_store.js#L153-L154

Comment (jcscottiii) [inline static_src/stores/user_store.js]: it's meant to be duplicated but let me re-work it so it's more clear as to why it's like that.

Comment (rememberlenny) [inline static_src/stores/user_store.js]: Got it

Comment (rememberlenny) [inline static_src/stores/user_store.js]: Looks good overall.

**Comment by rememberlenny** [comment]
@jcscottiii Merged because it does what it is supposed to do.

**Pull request merged**